docs/tutorials/hello-stacks.md

```diff
@@ -0,0 +1,759 @@
+          Connect Wallet
+        </button>
+      )}
+      <h1 className="text-6xl font-black">Hello Stacks</h1>
+      {userData && (
+        <div className="flex gap-4">
+          <input
+            className="p-4 border border-indigo-500 rounded"
+            placeholder="Write message here..."
+            onChange={handleMessageChange}
+            value={message}
+          />
+          <button
+            className="p-4 bg-indigo-500 rounded text-white"
+            onClick={submitMessage}
+          >
+            Submit New Message
+          </button>
+        </div>
+      )}
+      <div className="flex gap-4">
+        <input
+          className="p-4 border border-indigo-500 rounded"
+          placeholder="Paste transaction ID to look up message"
+          onChange={handleTransactionChange}
+          value={transactionId}
+        />
+        <button
+          className="p-4 bg-indigo-500 rounded text-white"
+          onClick={retrieveMessage}
+        >
+          Retrieve Message
+        </button>
+      </div>
+      {currentMessage.length > 0 ? (
+        <p className="text-2xl">{currentMessage}</p>
+      ) : (
+        ""
+      )}
+    </div>
+  );
+}
+
+export default App;
+```
+
+If you click 'Confirm' you should see the transaction id logged in the console and you can see the transaction in the block explorer.
+
+![new transaction](./tx-proof.png)
+
+## Retrieving Messages
+
+Now that we can write a message to the chain, we also want to add functionality so that we can retrieve a message from the chain.
+
+We'll do that by having users enter in a transaction ID and we'll query the chain for that transaction and the event data associated with it.
+
+The Stacks API is perfect for this, and we can query our local devnet chain directly.
+
+Let's set that up in our `retrieveMessage` function.
+
+```jsx
+const retrieveMessage = async () => {
+  const retrievedMessage = await fetch(
+    "http://localhost:3999/extended/v1/tx/events?" +
+      new URLSearchParams({
+        tx_id: transactionId,
+      })
+  );
+  const responseJson = await retrievedMessage.json();
+  setCurrentMessage(responseJson.events[0].contract_log.value.repr);
+};
+```
+
+Here we are using `fetch` to call the Stacks API and passing our transaction as a query parameter to get the events and then digging down through that returned data to get the printed message.
+
+You can view the documentation for this specific API call on [Hiro's website](https://docs.hiro.so/api#tag/Transactions/operation/get_filtered_events).
+
+If you run this you should see the message printed at the bottom of the screen.
+
+## Wrapping Up and Next Steps
+
+Here we've just gone over a very simple and very brief example of a Stacks application. This was meant to give you a high level introduction to the Stacks ecosystem and how you might begin building Stacks dapps.
+
+There is obvisouly a lot more to learn, so here are a few good places to continue your learning.
```

Comment:
Incorrect: obvisouly
Correct: obviously
(sorry I don't know how to use Github properly so I'm not sure if I'm submitting these proposed changes correctly, please do tell me if there a better way you wish me to submit)